runtime/gocont.go

```diff
@@ -129,6 +129,10 @@ func (c *GoCont) DebugInfo() *DebugInfo {
 	}
 }
 
+func (c *GoCont) Cleanup(t *Thread, err *Error) *Error {
+	return err
```

Comment:
Is this supposed to err here?

Reply:
Yeah, when you clean up it may be because there was an error.  From testing with C Lua, it seems that either the original error is returned, or some of the cleanup methods return an error (which then supersedes the original error).